### ISSUE TITLE
Dependency issue with breseq's recipe

### ISSUE THREAD

**Issue by ysu20** [bioconda/bioconda-recipes#6163]
The breseq recipe lists bowtie2 as a run time dependency.  However, when it is deployed in a virtual environment, it doesn't seem to deploy bowtie2's run time dependencies that are listed in bowtie2's recipe in the same virtual environment.  As a result, breseq doesn't work properly in the virtual environment, when it tries to do any alignment work by using bowtie2.

The workaround is to deploy bowtie2 in the virtual environment as well.  However it won't work for the breseq bio container, which is also based on this recipe.

**Comment by mbargull** [comment]
As @lparsons pointed out in #6637, this is likely also be caused by Conda picking up an older (broken) build of `bowtie2`.
@bioconda/core : to fix this, we could/should flag `bowtie2=2.2.5-0` as broken since it misses its run dependencies and newer builds are available. When we avoid that build, everything should work fine, see:
<details><summary><code>
$ conda create --dry-run -n x -c conda-forge -c bioconda breseq 'bowtie2!=2.2.5-0'
</code></summary><p>

```
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/maba/code/miniconda3/envs/x:

The following NEW packages will be INSTALLED:

    bowtie2:         2.3.3.1-py36pl5.22.0_0 bioconda   
    breseq:          0.29.0-r3.4.1_0        bioconda   
    bzip2:           1.0.6-1                conda-forge
    ca-certificates: 2017.7.27.1-0          conda-forge
    cairo:           1.14.6-5               conda-forge
    certifi:         2017.7.27.1-py36_0     conda-forge
    curl:            7.55.1-0               conda-forge
    fontconfig:      2.12.1-5               conda-forge
    freetype:        2.7-2                  conda-forge
    gettext:         0.19.7-1               conda-forge
    glib:            2.51.4-0               conda-forge
    graphite2:       1.3.9-0                conda-forge
    gsl:             2.1-2                  conda-forge
    harfbuzz:        1.4.3-0                conda-forge
    icu:             58.1-1                 conda-forge
    jpeg:            9b-2                   conda-forge
    krb5:            1.14.2-0               conda-forge
    libffi:          3.2.1-3                conda-forge
    libgcc:          5.2.0-0                           
    libiconv:        1.14-4                 conda-forge
    libpng:          1.6.28-2               conda-forge
    libssh2:         1.8.0-2                conda-forge
    libtiff:         4.0.7-1                conda-forge
    libxml2:         2.9.5-1                conda-forge
    ncurses:         5.9-10                 conda-forge
    openssl:         1.0.2l-0               conda-forge
    pango:           1.40.4-0               conda-forge
    pcre:            8.39-0                 conda-forge
    perl:            5.22.0.1-0             conda-forge
    pip:             9.0.1-py36_0           conda-forge
    pixman:          0.34.0-0               conda-forge
    python:          3.6.3-1                conda-forge
    r-base:          3.4.1-1                conda-forge
    readline:        6.2-0                  conda-forge
    setuptools:      36.6.0-py36_1          conda-forge
    sqlite:          3.13.0-1               conda-forge
    tbb:             2018_20170919-0        conda-forge
    tk:              8.5.19-2               conda-forge
    wheel:           0.30.0-py_1            conda-forge
    xz:              5.2.3-0                conda-forge
    zlib:            1.2.11-0               conda-forge
```
</p></details>

**Comment by bgruening** [comment]
@mbargull I wanted to label it as broken but there is no `bowtie2=2.2.5` with build number `0`?

**Comment by mbargull** [comment]
https://anaconda.org/bioconda/bowtie2/files/modal/info/568e9b60d45f8a6f1c1cd743
https://anaconda.org/bioconda/bowtie2/files/modal/info/56eeb6ed147ebb54771ce78c

**Comment by bgruening** [comment]
Anaconda has pagination, who knows that ;)
I marked them as broken.

**Comment by mbargull** [comment]
:D thanks!

**Comment by mbargull** [comment]
Now everything works as expected:
<details><summary><code>
$  conda create --dry-run -n x -c conda-forge -c bioconda breseq
</code></summary><p>

```
Fetching package metadata .........
Solving package specifications: .

Package plan for installation in environment /home/maba/code/miniconda3/envs/x:

The following NEW packages will be INSTALLED:

    bowtie2:         2.3.3.1-py36pl5.22.0_0 bioconda   
    breseq:          0.29.0-r3.4.1_0        bioconda   
    bzip2:           1.0.6-1                conda-forge
    ca-certificates: 2017.7.27.1-0          conda-forge
    cairo:           1.14.6-5               conda-forge
    certifi:         2017.7.27.1-py36_0     conda-forge
    curl:            7.55.1-0               conda-forge
    fontconfig:      2.12.1-5               conda-forge
    freetype:        2.7-2                  conda-forge
    gettext:         0.19.7-1               conda-forge
    glib:            2.51.4-0               conda-forge
    graphite2:       1.3.9-0                conda-forge
    gsl:             2.1-2                  conda-forge
    harfbuzz:        1.4.3-0                conda-forge
    icu:             58.1-1                 conda-forge
    jpeg:            9b-2                   conda-forge
    krb5:            1.14.2-0               conda-forge
    libffi:          3.2.1-3                conda-forge
    libgcc:          5.2.0-0                           
    libiconv:        1.14-4                 conda-forge
    libpng:          1.6.28-2               conda-forge
    libssh2:         1.8.0-2                conda-forge
    libtiff:         4.0.7-1                conda-forge
    libxml2:         2.9.5-1                conda-forge
    ncurses:         5.9-10                 conda-forge
    openssl:         1.0.2l-0               conda-forge
    pango:           1.40.4-0               conda-forge
    pcre:            8.39-0                 conda-forge
    perl:            5.22.0.1-0             conda-forge
    pip:             9.0.1-py36_0           conda-forge
    pixman:          0.34.0-0               conda-forge
    python:          3.6.3-1                conda-forge
    r-base:          3.4.1-1                conda-forge
    readline:        6.2-0                  conda-forge
    setuptools:      36.6.0-py36_1          conda-forge
    sqlite:          3.13.0-1               conda-forge
    tbb:             2018_20170919-0        conda-forge
    tk:              8.5.19-2               conda-forge
    wheel:           0.30.0-py_1            conda-forge
    xz:              5.2.3-0                conda-forge
    zlib:            1.2.11-0               conda-forge
```
</p></details>

@ysu20: can you confirm?

**Comment by ysu20** [comment]
I can confirm that bowtie2 is now deployed correctly as a dependency when deploying breseq.  However, it exposes a bug in breseq, which fails to parse bowtie2's version number correctly.  I am getting the following error when using breseq:
`
---> ERROR Required executable "bowtie2" version 2.0.0-beta7 or later not found.
---> Your version is 2.3.3.1.
---> See http://bowtie-bio.sourceforge.net/bowtie2
---> R        :: version 3.4.1 [/home/sund/conda-envs/breseq/bin/R]
`
Checking breseq's source code, I found that it expects bowtie2's version number encoded in triplets of numbers.  The workaround for resolving this issue is downgrading bowtie2 to 2.3.2.

To deploy breseq reliably, I suggest fixing bowtie2 to version 2.3.2 in breseq's runtime dependency until breseq is fixed properly. @mbargull : what do you think?

**Comment by mbargull** [comment]
Thanks for trying it out, @ysu20! Apparently there is a newer version of `breseq` which explicitly fixes that version parsing issue: https://github.com/barricklab/breseq/releases/v0.31.1
I'll just update the recipe to that version.

**Comment by ysu20** [comment]
That's great news. Thanks @mbargull. I will test the new build when it is available.

**Comment by mbargull** [comment]
Fixed with `conda 4.3.33` and `conda 4.4.8`.
